src/open_producten/producttypen/models/producttype.py

```diff
@@ -7,19 +7,51 @@
 from markdownx.models import MarkdownxField
 
 from open_producten.locaties.models import Contact, Locatie, Organisatie
+from open_producten.utils.fields import ChoiceArrayField
 from open_producten.utils.models import BasePublishableModel
 
 from .thema import Thema
 from .upn import UniformeProductNaam
 
 
+class ProductStateChoices(models.TextChoices):
+
+    INITIEEL = "initieel", _("Initieel")
+    GEREED = "gereed", _("Gereed")
+    ACTIEF = "actief", _("Actief")
+    INGETROKKEN = "ingetrokken", _("Ingetrokken")
+    GEWEIGERD = "geweigerd", _("Geweigerd")
+    VERLOPEN = "verlopen", _("Verlopen")
+
+
 class ProductType(BasePublishableModel):
     naam = models.CharField(
         verbose_name=_("product type naam"),
         max_length=100,
         help_text=_("naam van het product type."),
     )
 
+    code = models.CharField(
+        verbose_name=_("code"),
+        max_length=100,
+        help_text=_("code van het product type."),
+        unique=True,
+    )
+
+    toegestane_statussen = ChoiceArrayField(
+        models.CharField(
+            choices=[
+                choice
+                for choice in ProductStateChoices.choices
+                if choice[0] != ProductStateChoices.INITIEEL.value
```

Comment:
```suggestion
                if choice[0] != ProductStateChoices.INITIEEL
```
Value is unnecessary field (kinda defeats the point of an ENUM)

Reply:
Also I don't particularly like the list comprehension here but it's fine.